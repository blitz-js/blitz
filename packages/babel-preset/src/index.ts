import type { TransformOptions } from '@babel/core';
import AddBlitzAppRoot from './add-blitz-app-root';
import FixNodeFileTrace from './fix-node-file-trace';
import RewriteImports from './rewrite-imports';

// eslint-disable-next-line import/no-default-export
export default function preset(_api: any, options = {}) {
  // const isTest = _api.env('test');
  const isRunningInJest =
    process.env.JEST_WORKER_ID && !process.env.__NEXT_TEST_MODE;

  const config: TransformOptions = {
    presets: [[require('next/babel'), options]],
    plugins: [
      FixNodeFileTrace,
      [
        require('babel-plugin-superjson-next'),
        { exclude: ['dehydratedState'] },
      ],
      AddBlitzAppRoot,
    ],
  };

  if (!isRunningInJest) {
    config.plugins!.push(RewriteImports);
  }

  return config;
}
