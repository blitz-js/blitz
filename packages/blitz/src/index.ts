export * from '@blitzjs/core'
export {withBlitz} from '@blitzjs/server'
