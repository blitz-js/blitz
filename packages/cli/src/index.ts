export const sum = (a: number, b: number) => {
  if ('development' === process.env.NODE_ENV) {
    console.log('boop');
  }
  return a + b;
};
