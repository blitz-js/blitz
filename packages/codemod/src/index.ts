import arg from "arg"

const commonArgs = {
  // Types
  "--help": Boolean,

  // Aliases
  "-h": "--help",
}

const args = arg(commonArgs, {
  permissive: true,
})

const commands: {[command: string]: () => Promise<() => void>} = {
  "upgrade-legacy": () => import("./upgrade-legacy").then((i) => i.upgradeLegacy),
}

const foundCommand = Boolean(commands[args._[0] as string])
const forwardedArgs = foundCommand ? args._.slice(1) : args._

if (foundCommand) {
  commands[args["_"][0] as string]?.()
    .then((exec: any) => exec(forwardedArgs))
    .catch((err) => {
      console.log(err)
    })
} else {
  console.log("Codemod not found. Try one of these:")
  console.log(`${Object.keys(commands).map((c) => c)}`)
}
