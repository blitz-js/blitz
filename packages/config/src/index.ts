import pkgDir from "pkg-dir"
import {join} from "path"
import {existsSync} from "fs"

const configFiles = ["blitz.config.js", "next.config.js"]
/**
 * @param {boolean | undefined} reload - reimport config files to reset global cache
 */
export const getConfig = (reload?: boolean): Record<string, unknown> => {
  if (global.blitzConfig && Object.keys(global.blitzConfig).length > 0 && !reload) {
    return global.blitzConfig
  }

  let blitzConfig = {}
  const projectRoot = pkgDir.sync() || process.cwd()

  for (const configFile of configFiles) {
    if (existsSync(join(projectRoot, configFile))) {
      const path = join(projectRoot, configFile)
      const file = require(path)
      let contents
      if (typeof file === "function") {
        contents = file()
      } else {
        contents = file
      }
      blitzConfig = {
        ...blitzConfig,
        ...contents,
      }
    }
  }

  global.blitzConfig = blitzConfig

  return blitzConfig
}
