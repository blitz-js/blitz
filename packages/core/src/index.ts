export * from './useQuery'
export * from './ssrQuery'
export * from './rpc'

export {
  GetStaticProps,
  GetStaticPaths,
  GetServerSideProps,
  NextPage as BlitzPage,
  NextApiRequest as BlitzApiRequest,
  NextApiResponse as BlitzApiResponse,
} from 'next'

export {default as Head} from 'next/head'

export {default as Link} from 'next/link'

export {Router, useRouter, withRouter} from 'next/router'

export {
  default as Document,
  Html,
  Head as DocumentHead,
  Main,
  NextScript,
  DocumentContext,
} from 'next/document'

export {default as dynamic} from 'next/dynamic'
