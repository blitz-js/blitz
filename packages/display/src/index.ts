import chalk from 'chalk'
import ora from 'ora'
import readline from 'readline'

// const blitzTrueBrandColor = '6700AB'
const blitzBrightBrandColor = '8a3df0'

// Using brigh brand color so it's better for dark terminals
const brandColor = blitzBrightBrandColor

const withBrand = (str: string) => {
  return chalk.hex(brandColor).bold(str)
}

const withWarning = (str: string) => {
  return `⚠️  ${chalk.yellow(str)}`
}

const withCaret = (str: string) => {
  return `${chalk.gray('>')} ${str}`
}

const withCheck = (str: string) => {
  return `${chalk.green('✔')} ${str}`
}

const withX = (str: string) => {
  return `${chalk.red.bold('✕')} ${str}`
}

/**
 * Logs a branded purple message to stdout.
 *
 * @param {string} msg
 */
const branded = (msg: string) => {
  console.log(chalk.hex(brandColor).bold(msg))
}

/**
 * Clears the line and optionally log a message to stdout.
 *
 * @param {string} msg
 */
const clearLine = (msg?: string) => {
  readline.clearLine(process.stdout, 0)
  readline.cursorTo(process.stdout, 0)
  msg && process.stdout.write(msg)
}

/**
 * Logs a red error message to stderr.
 *
 * @param {string} msg
 */
const warning = (msg: string) => {
  console.log(withCaret(withWarning(msg)))
}

/**
 * Logs a red error message to stderr.
 *
 * @param {string} msg
 */
const error = (msg: string) => {
  console.error(withX(chalk.red.bold(msg)))
}

/**
 * Logs a subtle gray message to stdout.
 *
 * @param {string} msg
 */
const meta = (msg: string) => {
  console.log(withCaret(chalk.gray(msg)))
}

/**
 * Logs a progress message to stdout.
 *
 * @param {string} msg
 */
const progress = (msg: string) => {
  console.log(withCaret(chalk.bold(msg)))
}

const info = (msg: string) => {
  console.log(chalk.bold(msg))
}

const spinner = (str: string) => {
  return ora({
    text: str,
    color: 'blue',
    spinner: {
      interval: 120,
      frames: ['◢', '◣', '◤', '◥'],
    },
  })
}

/**
 * Logs a green success message to stdout.
 *
 * @param {string} msg
 */
const success = (msg: string) => {
  console.log(withCheck(chalk.green(msg)))
}

/**
 * Colorizes a variable for display.
 *
 * @param {string} val
 */
const variable = (val: any) => {
  return chalk.cyan.bold(`${val}`)
}

/**
 * If the DEBUG env var is set this will write to the console
 * @param str msg
 */
const debug = (str: string) => {
  process.env.DEBUG && console.log(str)
}

export const log = {
  withBrand,
  withWarning,
  withCaret,
  withCheck,
  withX,
  branded,
  clearLine,
  error,
  warning,
  meta,
  progress,
  spinner,
  success,
  variable,
  info,
  debug,
}
