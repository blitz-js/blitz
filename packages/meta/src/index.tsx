import * as React from 'react'

const makeTitle = (title: string, name: string) => (title === name ? title : `${title} â€“ ${name}`)

export const Meta = ({
  as: ElementType = React.Fragment,
  name = 'Blitz', // site name
  title = 'Blitz', // page title
  description,
  image,
  color = '#6700eb',
  manifest = 'https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/site.webmanifest',
}: {
  as?: any
  name?: string
  title?: string
  description?: string
  image?: string
  color?: string
  manifest?: string
}) => (
  <ElementType>
    <meta key="og_locale" property="og:locale" content="en_US" />
    <meta key="og_type" property="og:type" content="website" />
    <meta key="og_site" property="og:site_name" content={name} />
    <meta key="tw_site" name="twitter:site" content="@blitz_js" />
    <title key="title">{makeTitle(title, name)}</title>
    <meta key="og_title" property="og:title" content={makeTitle(title, name)} />
    <meta key="tw_title" name="twitter:title" content={makeTitle(title, name)} />
    {description && (
      <>
        <meta key="desc" name="description" content={description} />
        <meta key="og_desc" property="og:description" content={description} />
        <meta key="tw_desc" name="twitter:description" content={description} />
      </>
    )}
    {image && (
      <>
        <meta key="og_img" property="og:image" content={image} />
        <meta key="tw_card" name="twitter:card" content="summary_large_image" />
        <meta key="tw_img" name="twitter:image" content={image} />
      </>
    )}
    <meta key="theme_color" name="theme-color" content={color} />
    <meta key="tile_color" name="msapplication-TileColor" content={color} />
    <link
      key="safari_icon"
      rel="mask-icon"
      href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/safari-pinned-tab.svg"
      color={color}
    />
    <link
      key="apple_icon"
      rel="apple-touch-icon"
      sizes="180x180"
      href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/apple-touch-icon.png"
    />
    <link
      key="favicon_32"
      rel="icon"
      type="image/png"
      sizes="32x32"
      href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-32x32.png"
    />
    <link
      key="favicon_16"
      rel="icon"
      type="image/png"
      sizes="16x16"
      href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-16x16.png"
    />
    <link key="manifest" rel="manifest" href={manifest} />
  </ElementType>
)
