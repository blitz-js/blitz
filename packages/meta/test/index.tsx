import * as React from 'react'
import {render, cleanup} from '@testing-library/react'

import {Meta} from '../src'

afterEach(cleanup)

test('Meta renders', () => {
  const {container, getByText} = render(<Meta />)
  expect(getByText('Blitz')).toBeTruthy()
  expect(container.querySelector('[property="og:title"][content="Blitz"]')).toBeTruthy()
  expect(container.querySelector('[name="twitter:title"][content="Blitz"]')).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('Meta renders custom title', () => {
  const title = 'Custom Title â€“ Blitz'
  const {container, getByText} = render(<Meta title="Custom Title" />)
  expect(getByText(title)).toBeTruthy()
  expect(container.querySelector(`[property="og:title"][content="${title}"]`)).toBeTruthy()
  expect(container.querySelector(`[name="twitter:title"][content="${title}"]`)).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('Meta renders image', () => {
  const url = 'https://via.placeholder.com/1024x512'
  const {container} = render(<Meta image={url} />)
  expect(container.querySelector('meta[name="twitter:card"][content="summary_large_image"]')).toBeTruthy()
  expect(container.querySelector(`[content="${url}"]`)).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('Meta renders custom color', () => {
  const color = '#fff'
  const {container} = render(<Meta color={color} />)
  expect(container.querySelector(`meta[name="theme-color"][content="${color}"]`)).toBeTruthy()
  expect(container).toMatchSnapshot()
})
