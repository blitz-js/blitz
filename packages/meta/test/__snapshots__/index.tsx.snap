// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Blitz"
    property="og:site_name"
  />
  <meta
    content="@blitz_js"
    name="twitter:site"
  />
  <title>
    Blitz
  </title>
  <meta
    content="Blitz"
    property="og:title"
  />
  <meta
    content="Blitz"
    name="twitter:title"
  />
  <meta
    content="#6700eb"
    name="theme-color"
  />
  <meta
    content="#6700eb"
    name="msapplication-TileColor"
  />
  <link
    color="#6700eb"
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders custom color 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Blitz"
    property="og:site_name"
  />
  <meta
    content="@blitz_js"
    name="twitter:site"
  />
  <title>
    Blitz
  </title>
  <meta
    content="Blitz"
    property="og:title"
  />
  <meta
    content="Blitz"
    name="twitter:title"
  />
  <meta
    content="#fff"
    name="theme-color"
  />
  <meta
    content="#fff"
    name="msapplication-TileColor"
  />
  <link
    color="#fff"
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders custom title 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Blitz"
    property="og:site_name"
  />
  <meta
    content="@blitz_js"
    name="twitter:site"
  />
  <title>
    Custom Title – Blitz
  </title>
  <meta
    content="Custom Title – Blitz"
    property="og:title"
  />
  <meta
    content="Custom Title – Blitz"
    name="twitter:title"
  />
  <meta
    content="#6700eb"
    name="theme-color"
  />
  <meta
    content="#6700eb"
    name="msapplication-TileColor"
  />
  <link
    color="#6700eb"
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/site.webmanifest"
    rel="manifest"
  />
</div>
`;

exports[`Meta renders image 1`] = `
<div>
  <meta
    content="en_US"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Blitz"
    property="og:site_name"
  />
  <meta
    content="@blitz_js"
    name="twitter:site"
  />
  <title>
    Blitz
  </title>
  <meta
    content="Blitz"
    property="og:title"
  />
  <meta
    content="Blitz"
    name="twitter:title"
  />
  <meta
    content="https://via.placeholder.com/1024x512"
    property="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="https://via.placeholder.com/1024x512"
    name="twitter:image"
  />
  <meta
    content="#6700eb"
    name="theme-color"
  />
  <meta
    content="#6700eb"
    name="msapplication-TileColor"
  />
  <link
    color="#6700eb"
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="https://raw.githubusercontent.com/blitz-js/blitz/canary/favicons/site.webmanifest"
    rel="manifest"
  />
</div>
`;
